csrc/multidevice/communicator.h

```diff
@@ -31,8 +31,8 @@ namespace nvfuser {
 
 using RankType = DeviceIdxType;
 
-// Supported backends. TODO: only tested with nccl for now
-enum class CommunicatorBackend { nccl, ucc, gloo };
+// Supported backends. TODO: gloo untested
+enum class CommunicatorBackend { nccl, ucc, gloo, none };
 
 constexpr CommunicatorBackend comm_backend_default = CommunicatorBackend::nccl;
```

Comment:
What about falling back to ucc if nccl is not available ? This can still be determined at compile time, with something like:
```
#ifdef USE_C10D_NCCL
constexpr CommunicatorBackend comm_backend_default = CommunicatorBackend::nccl;
#elif USE_C10D_UCC
constexpr CommunicatorBackend comm_backend_default = CommunicatorBackend::ucc;
#endif
```